graphjet-core/src/main/java/com/twitter/graphjet/algorithms/PageRank.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2016 Twitter. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twitter.graphjet.algorithms;
+
+import com.twitter.graphjet.bipartite.api.EdgeIterator;
+import com.twitter.graphjet.directed.api.OutIndexedDirectedGraph;
+import it.unimi.dsi.fastutil.longs.LongArrayList;
+import it.unimi.dsi.fastutil.longs.LongIterator;
+import it.unimi.dsi.fastutil.longs.LongOpenHashSet;
+
+/**
+ * <p>
+ * Implementation of PageRank. This implementation initializes an array of doubles to hold the
```

Comment:
Sorry I'm a bit confused about the model here: for running pagerank, are we assuming that the graph has a fixed set of nodes? How about edges? The reason I want to clarify this is that this will determine the API to GraphJet.